packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -73,30 +73,66 @@ function preload(
       if (dep in seen) return
       // @ts-ignore
       seen[dep] = true
+      const seperatorIdx = dep.lastIndexOf('/')
+      const shortDep = seperatorIdx >= 0 ? dep.slice(seperatorIdx + 1) : dep
       const isCss = dep.endsWith('.css')
       const cssSelector = isCss ? '[rel="stylesheet"]' : ''
       // @ts-ignore check if the file is already preloaded by SSR markup
-      if (document.querySelector(`link[href="${dep}"]${cssSelector}`)) {
-        return
+      const possibleLinks = document.querySelectorAll<HTMLLinkElement>(
+        `link[href$="${shortDep}"]${cssSelector}`
+      )
+      for (let i = 0; i < possibleLinks.length; ++i) {
+        const currentPath = possibleLinks[i].href
+        if (
+          currentPath === dep ||
+          new URL(currentPath, importerUrl).href === dep
```

Comment:
Unless I'm missing something I think that line 88 can be safely removed. Here's my reasoning:

`possibleLinks[i].href` is an absolute URL already because this expression accesses the href _property_ (not an _attribute_) and browsers must normalize all URL containing _properties_ to be absolute. See [here](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflecting-content-attributes-in-idl-attributes:idl-domstring-5) under "If the content attribute is defined to contain a URL". Because of that `currentPath` also must be absolute. Doing `new URL(AN_ABSOLUTE_URL, whatever).href` returns `AN_ABSOLUTE_URL` so 

```
// original code
          currentPath === dep ||
          new URL(currentPath, importerUrl).href === dep

// given that currentPath is absolute is equivalent to
          currentPath === dep ||
          currentPath === dep

// which is equivalent to
          currentPath === dep
```

Reply:
Tnx, just used your code instead.